### ISSUE TITLE
[Dev]: Skip Test-Stage Teardown When Tests Fail

### ISSUE THREAD

**Issue by medelman17** [jaredpalmer/tsdx#239]
### Current Behavior

`Tsdx's` [testing suite](https://github.com/jaredpalmer/tsdx/tree/master/test/tests) works, in part, by loading in and out different [configuration/template-fixtures](https://github.com/jaredpalmer/tsdx/tree/master/test/fixtures) and using the build under test to generate relevant test cases. These test cases are then linted, prettier'd, and otherwise-parsed. 

Unsurprisingly, this process generates files that are stored on disk. Thus, `tsdx` takes care at the end of each test to clean up after itself. However, because this cleanup happens regardless of whether a given test passes, the behavior has the unfortunate consequence of also cleaning up the best evidence for why the test may have failed. 

### Desired Behavior

As a `tsdx` contributor, I should be given as much context as is possible from which to determine why my tests are failing--thus, `tsdx` should preserve, rather than tear-down the environments for failing tests.  

### Suggested Solution

I want to look into whether our test-runner is or can be made aware of whether a given test has failed in its `afterEach` lifecycle hook. If so, ideally we could just skip teardown for the ones that failed. We already guard against dirty environments by running the same clean-up process `beforeAll`. 

Unfortunately, this likely wouldn't be enough because, as it stands, we generate multiple sets of files (based on different build variations) per fixture. And these files are generated and temporarily stored in the same location--hence, the need for the `afterEach` vs `afterAll` cleanup. As a result, we would likely have to ensure that each of the various sets were generated in unique locations. 

Worst case scenario--i.e., it cannot be known whether a given test passed or failed prior to teardown--the solution would be the same as the above, with the exception that (1) we would be leaving more junk for a longer period of time and (2) we would change the `beforeAll` to a `beforeEach` and eliminate the `afterEach` teardown.  
 
### Who does this impact? Who is this for?

This issue/feature-request only affects `tsdx` contributors.  

### Describe alternatives you've considered

Well, I've tried the following, neither of which are awesome:  

1. re-running the failing test and hoping to kill it at just the right moment after the relevant test starts but before its garbage is collected; and

2. recreating the relevant test setup by hand on the command line.

**Comment by jaredpalmer** [comment]
Yes to this. I copied this technique from razzle which copies it from nytimes/kyt. It’s several years old now and def needs a refresh
